models/gan.py

```diff
@@ -0,0 +1,300 @@
+import numpy as np
+import torch
+from torch import nn
+import torch.nn.functional as F
+try:
+    # relative import
+    from .base_models import BaseModelSRL, ConvSN2d, ConvTransposeSN2d, LinearSN, UNet
+    from .base_trainer import BaseTrainer
+    from ..losses.losses import ganNonSaturateLoss, autoEncoderLoss, ganBCEaccuracy, AEboundLoss
+except:
+    from models.base_models import BaseModelSRL, ConvSN2d, ConvTransposeSN2d, LinearSN, UNet
+    from models.base_trainer import BaseTrainer
+    from losses.losses import ganNonSaturateLoss, autoEncoderLoss, ganBCEaccuracy, AEboundLoss
+    
```

Comment:
Please update, missing docstring here and there. Also commented code to be removed.
I would suggest in case of specific architecture to include references when possible.